langkit/openai_wrapper.py

```diff
@@ -0,0 +1,53 @@
+import openai
+import contextlib
+import functools
+import whylogs as why
+from whylogs.experimental.core.udf_schema import udf_schema
+
+class RollingLogger:
+    def __init__(self):
+        self.logger = why.logger(mode="rolling", interval=5, when="M", base_name="langkit", schema=udf_schema())
+        self.logger.append_writer("whylabs")
+    
+    def log(self, dict):
+        self.logger.log(dict)
+    
+    def close(self):
+        self.logger.close()
+
+
+def log_openai_with_langkit(func, logger):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        response = func(*args, **kwargs)
+        messages = kwargs.get('messages')
+        if messages:
+            user_prompt = [message['content'] for message in messages if message['role'] == 'user'][0]
+            system_prompt = [message['content'] for message in messages if message['role'] == 'system'][0]
```

Comment:
```suggestion
            system_prompt = [message['content'] for message in messages if message['role'] == 'system'][0]
            llm_response = ""
```
then you can drop the else clauses for simpler code.